client/src/board/boardwidget.cpp

```diff
@@ -778,6 +768,61 @@ void BoardWidget::rightClicked()
     rotateTile(true);
 }
 
+void BoardWidget::startHoldTimer(const std::shared_ptr<NXButtonRailTile>& nxButton)
+{
+  stopTimerAndReleaseButtons();
+
+  m_releaseButton1 = nxButton;
+
+  m_timer = new QTimer(this);
```

Comment:
I suggest to add a `assert(m_timer);` here, to check if `m_timer` is `nullptr` as expected.